scripts/sct_crop_image.py

```diff
@@ -86,7 +86,7 @@ def crop(self):
                 sct.printv('WARNING: Mask was specified for cropping. Arguments -start and -end will be ignored', 1, 'warning')
             self.start, self.end = find_mask_boundaries(self.mask)
             # overwrite dim to specify 3 dimensions
-            self.dim = [0, 1, 2]
+            self.dim = range(len(Image(self.mask).data.shape))
```

Comment:
You already calculate the dim in find_mask_boundaries. You could return the dim value and set it to self.dim

This avoids inconsistencies in the future.